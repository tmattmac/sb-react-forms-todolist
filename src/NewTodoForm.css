.NewTodoForm {
    display: flex;
    flex-flow: row wrap;
}

.NewTodoForm input {
    flex-grow: 2;
}