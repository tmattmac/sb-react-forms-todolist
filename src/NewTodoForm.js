import React, { useState, useRef } from 'react';
import './NewTodoForm.css';

function NewTodoForm({ addTodo }) {
    const [formData, setFormData] = useState({ text: '' });
    const form = useRef(null);
    
    const handleChange = e => {
        const { name, value } = e.target;
        setFormData(formData => ({ ...formData, [name]: value }));
    };

    const handleSubmit = e => {
        e.preventDefault();
        addTodo(formData);
        setFormData({ text: '' });
    };

    return (
        <form className="NewTodoForm" onSubmit={handleSubmit} ref={form} data-testid="form">
            <input
                type="text"
                placeholder="Add todo text here"
                name="text"
                value={formData.text}
                onChange={handleChange}
                required
            />
            <button>Add</button>
        </form>
    );
}

export default NewTodoForm;
