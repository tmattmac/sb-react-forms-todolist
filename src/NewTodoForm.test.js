import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test('renders', () => {
    render(<NewTodoForm />);
});

test('render matches snapshot', () => {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

test('submitting data calls callback with data', () => {
    const mockAddTodo = jest.fn();
    const { getByText, getByTestId } = render(<NewTodoForm addTodo={mockAddTodo} />);
    const form = getByTestId('form');
    const input = form.querySelector('input');
    fireEvent.change(input, { target: { value: "Test Todo" } });
    fireEvent.submit(form);
    
    expect(mockAddTodo).toHaveBeenCalledWith(
        expect.objectContaining({ text: "Test Todo" })
    );
});