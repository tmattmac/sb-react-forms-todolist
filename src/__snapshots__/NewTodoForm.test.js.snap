// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
    data-testid="form"
  >
    <input
      name="text"
      placeholder="Add todo text here"
      required=""
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
